src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -16,49 +18,454 @@
-    }
+    mockConversationStore = Mockito.mock(ConversationStore.class);
+    adminServlet.setConversationStore(mockConversationStore);
+
+    mockMessageStore = Mockito.mock(MessageStore.class);
+    adminServlet.setMessageStore(mockMessageStore);
+
+  }
+
+  @Test
+  public void testDoGet_NonAdmin() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("not an admin");
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockResponse).sendError(HttpServletResponse.SC_NOT_FOUND);
+  }
+
+  @Test
+  public void testDoGet_Admin() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  // Number of users tests
+  @Test
+  public void testNumberOfUsers_NoUsers() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    List<User> mockAllUsers = new ArrayList<User>();
+
+    Mockito.when(mockUserStore.getAllUsers()).thenReturn(mockAllUsers);
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("numberOfUsers", 0);
+
+  }
+
+  @Test
+  public void testNumberOfUsers_MultipleUsers() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    User u1 = new User(UUID.randomUUID(), "user1", "hashed_pw", Instant.now());
+    User u2 = new User(UUID.randomUUID(), "user2", "hashed_pw", Instant.now());
+    User u3 = new User(UUID.randomUUID(), "user3", "hashed_pw", Instant.now());
+
+    List<User> mockAllUsers = new ArrayList<User>();
+    mockAllUsers.add(u1);
+    mockAllUsers.add(u2);
+    mockAllUsers.add(u3);
+
+    Mockito.when(mockUserStore.getAllUsers()).thenReturn(mockAllUsers);
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("numberOfUsers", 3);
+  }
+
+  // Newest user tests
+  @Test
+  public void testNewestUser_NoUsers() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    List<User> mockAllUsers = new ArrayList<User>();
+    Mockito.when(mockUserStore.getAllUsers()).thenReturn(mockAllUsers);
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("newestUser", "N/A");
+  }
+
+  @Test
+  public void testNewestUser_MultipleUsers() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    User u1 = new User(UUID.randomUUID(), "user1", "hashed_pw", Instant.now());
+    User u2 = new User(UUID.randomUUID(), "user2", "hashed_pw", Instant.now());
+    User u3 = new User(UUID.randomUUID(), "user3", "hashed_pw", Instant.now());
```

Comment:
See my comment in the code under test above--these users all have (approximately) the same creation timestamp. What happens if you manually give them different values? (`Instant.ofEpochSecond()` might be useful here.)